postgresql/postgresql_client.go

```diff
@@ -0,0 +1,167 @@
+		},
+	)
+	if err != nil {
+		return server, fmt.Errorf("cannot update pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return server, fmt.Errorf("cannot get the pg server update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// DeleteServer deletes the PostgreSQL server.
+func DeleteServer(ctx context.Context, serversClient pg.ServersClient, serverName string) (resp autorest.Response, err error) {
+
+	future, err := serversClient.Delete(ctx, config.GroupName(), serverName)
+	if err != nil {
+		return resp, fmt.Errorf("cannot delete the pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return resp, fmt.Errorf("cannot get the pg server update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// GetFwRulesClient returns the FirewallClient
+func GetFwRulesClient() pg.FirewallRulesClient {
+	fwrClient := pg.NewFirewallRulesClient(config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
+	fwrClient.Authorizer = a
+	fwrClient.AddToUserAgent(config.UserAgent())
+	return fwrClient
+}
+
+// CreateOrUpdateFirewallRule given the firewallname and new properties it updates the firewall rule.
+func CreateOrUpdateFirewallRule(ctx context.Context, fwrClient pg.FirewallRulesClient, serverName, firewallRuleName, startIPAddr, endIPAddr string) error {
+
+	_, err := fwrClient.CreateOrUpdate(
+		ctx,
+		config.GroupName(),
+		serverName,
+		firewallRuleName,
+		pg.FirewallRule{
+			FirewallRuleProperties: &pg.FirewallRuleProperties{
+				StartIPAddress: &startIPAddr,
+				EndIPAddress:   &endIPAddr,
+			},
+		},
+	)
+
+	return err
+}
+
+// GetConfigurationsClient creates and returns the configuration client for the server.
+func GetConfigurationsClient() pg.ConfigurationsClient {
+	configClient := pg.NewConfigurationsClient(config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
+	configClient.Authorizer = a
+	configClient.AddToUserAgent(config.UserAgent())
+	return configClient
+}
+
+// GetConfiguration given the server name and configuration name it returns the configuration.
+func GetConfiguration(ctx context.Context, configClient pg.ConfigurationsClient, serverName string, configurationName string) (pg.Configuration, error) {
+
+	// Get the configuration.
+	configuration, err := configClient.Get(ctx, config.GroupName(), serverName, configurationName)
+
+	if err != nil {
+		return configuration, fmt.Errorf("cannot get the configuration with name %s", configurationName)
+	}
+
+	return configuration, err
+}
+
+// UpdateConfiguration given the name of the configuation and the configuration object it updates the configuration for the given server.
+func UpdateConfiguration(ctx context.Context, configClient pg.ConfigurationsClient, serverName string, configurationName string, configuration pg.Configuration) (updatedConfig pg.Configuration, err error) {
+

```

Comment:
Extra empty line here, could we remove this?

Reply:
Fixed